docs/_future/custom_functions.md

```diff
@@ -0,0 +1,174 @@
+
+
+#### 2. Add function listener and handler(s)
+Once your function has been defined, you'll need to register a function listener to trigger the function's functionality as well as any other related events.
+
+To do this, create a file in `listeners/functions` directory for your function listener (in this case, the file will be called `hello-world.js`). This file will use the function definition you created earlier and also implement any needed additional logic&mdash;in this case, it'll send a message to greet someone based on what is being passed in as inputs. 
+
+```js
+// listeners/functions/hello-world.js
+// For more information about functions: https://api.slack.com/future/functions
+const { SlackFunction } = require('@slack/bolt');
+
+// Get our Function Definition from the manifest!
+const { SampleFunctionDefinition } = require('../../manifest/functions/sample-function');
+
+// Here is the work we want to do!
+const helloWorld = async ({ event, client, complete }) => {
+  const { recipient, channel, message } = event.inputs;
+  const salutations = ['Hello', 'Hi', 'Howdy', 'Hola', 'Salut'];
+  const salutation = salutations[Math.floor(Math.random() * salutations.length)];
+  try {
+    await client.chat.postMessage({
+      channel,
+      text: `${salutation}, <@${recipient}>! :wave: Someone sent the following greeting: \n\n>${message}`,
+    });
+    complete();
+  } catch (err) {
+    // Complete function with an error
+    await complete({ error: `There was an issue: ${err}` });
+    throw (err);
+  }
+};
+
+// Let's create a new Slack Function with helloWorld as its handler
+const helloWorldFunc = new SlackFunction(SampleFunctionDefinition.id, helloWorld);
+
+module.exports = { helloWorldFunc };
+```
+The above file declares a function handler, `helloWorld`, that takes in inputs from the `event`, which is the payload received when your function is being executed. It executes logic within the handler to send a message with a random greeting to the specified recipient in the desired channel. Then, a new `SlackFunction` instance is declared that actually links the `helloWorld` handler to the function being run through its function ID (`SampleFunctionDefinition`):
+```js
+// Let's create a new Slack Function with helloWorld as its handler
+const helloWorldFunc = new SlackFunction(SampleFunctionDefinition.id, helloWorld);
+```
+
+Once this file has been created, the function handler can be registered in an `index.js` file within the `listeners/functions` directory:
+```js
+// listeners/functions/index.js
+const { helloWorldFunc } = require('./hello-world');
+
+// Register a complete function
+module.exports.register = (app) => {
+  app.function(helloWorldFunc);
+  // Register another function here
+};
+```
+
+You can see the full completed `hello-world.js` function listener in our [Bolt for JavaScript Starter Template](https://github.com/slack-samples/bolt-js-starter-template/blob/future/listeners/functions/hello-world.js).
```

Comment:
```suggestion

```
@stevengill - forgot to add this comment to my review - we can delete this part since we're linking to the `hello-world.js` file above!